test/e2e/externalservices/externalservices_kubernetes_without_egress.go

```diff
@@ -30,124 +30,89 @@ spec:
 
-		// then communication outside of the Mesh works
-		Eventually(func(g Gomega) {
-			_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-				"curl", "-v", "-m", "3", "--fail", "http://externalservice-http-server.externalservice-namespace:10080")
-			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stderr).To(ContainSubstring("HTTP/1.1 200 OK"))
-		}, "1m", "3s").Should(Succeed())
+		// then communication outside of the Mesh works, because it goes through passthrough
+		Eventually(trafficAllowed, "30s", "1s").Should(Succeed())
 
-		// when passthrough is disabled on the Mesh and no egress
-		err = YamlK8s(fmt.Sprintf(meshDefaulMtlsOn, "false", "false"))(cluster)
-		Expect(err).ToNot(HaveOccurred())
+		// when passthrough is disabled
+		Expect(cluster.Install(YamlK8s(fmt.Sprintf(meshDefaulMtlsOn, "false", "false")))).To(Succeed())
 
-		// then communication outside of the Mesh works
-		Eventually(func(g Gomega) {
-			_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-				"curl", "-v", "-m", "3", "--fail", "http://externalservice-http-server.externalservice-namespace:10080")
-			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stderr).To(ContainSubstring("HTTP/1.1 200 OK"))
-		}, "1m", "3s").Should(Succeed())
+		// then accessing the external service is no longer possible
+		Eventually(trafficBlocked, "30s", "1s").Should(Succeed())
 
 		// when apply external service
-		err = YamlK8s(fmt.Sprintf(externalService,
-			es1, es1,
-			"externalservice-http-server.externalservice-namespace.svc.cluster.local", 10080, // .svc.cluster.local is needed, otherwise Kubernetes will resolve this to the real IP
-			"false"))(cluster)
-		Expect(err).ToNot(HaveOccurred())
+		Expect(cluster.Install(YamlK8s(externalService))).To(Succeed())
 
-		// and passthrough is disabled on the Mesh and zone egress enabled
-		err = YamlK8s(fmt.Sprintf(meshDefaulMtlsOn, "false", "true"))(cluster)
-		Expect(err).ToNot(HaveOccurred())
+		// then communication outside of the Mesh works, because it goes directly from the client to a service
+		Eventually(trafficAllowed, "30s", "1s").Should(Succeed())
 
-		// then accessing the external service is no longer possible
-		Eventually(trafficBlocked, "30s", "1s").Should(HaveOccurred())
-	})
-
-	It("should route to external-service over tls", func() {
-		// given Mesh with passthrough enabled and with zone egress
-		err := YamlK8s(fmt.Sprintf(meshDefaulMtlsOn, "true", "true"))(cluster)
-		Expect(err).ToNot(HaveOccurred())
-
-		// when apply external service
-		err = YamlK8s(fmt.Sprintf(externalService,
-			es2, es2,
-			"externalservice-https-server.externalservice-namespace.svc.cluster.local", 10080,
-			"true"))(cluster)
-		Expect(err).ToNot(HaveOccurred())
+		// when passthrough is disabled on the Mesh and zone egress enabled
+		Expect(cluster.Install(YamlK8s(fmt.Sprintf(meshDefaulMtlsOn, "false", "true")))).To(Succeed())
```

Comment:
It might be nit picking. But it seems that this line is the only place where we need this to run in e2e right?

Should the rest of the test move to `e2e_env/multizone` and remove this test altogether? This line seems to already be covered in `test/e2e/external_services/localityawarelb_multizone`

Reply:
Multizone checks this flow
client in zone 1 -> zone egress in zone 1 -> zone ingress zone 2 (A) -> zone egress zone 2 (B) -> external service

We test cases when A is down or when B is down that the client cannot hit external service.